zerver/views/auth.py

```diff
@@ -293,7 +297,14 @@ def maybe_send_to_registration(
             prereg_user.streams.set(streams_to_subscribe)
         if include_realm_default_subscriptions is not None:
             prereg_user.include_realm_default_subscriptions = include_realm_default_subscriptions
-        prereg_user.invited_as = invited_as
+
+        if role is not None:
+            # role, if specified as argument to this function,
+            # takes precedence over anything else, as it is an explicit
+            # statement of what the user should be created as.
```

Comment:
Maybe this one can just refer to the comment suggested above.

Reply:
Hmm, I moved and tweaked this elaboration to the docsting of `maybe_send_to_registration`:
```
    ....

    Important: role, if specified as argument to this function,
    takes precedence over anything else, as it is an explicit
    statement of what the user should be created as, and is likely
    being synced from some user management system tied to the
    authentication method used.
    """
```

and then tweaked this place to just briefly mention it:
```
        if role is not None:
            # As explained at the top of this function, role, if specified as argument,
            # takes precedence over the role implied by the invitation.
            prereg_user.invited_as = role
```

does this work?

Reply:
Yep!